File: bot/Plugins/Vera/modules/vera.rs
> object temperature python
    import urllib, json
    url = "http://192.168.1.100:3480/data_request?id=status&output_format=json&DeviceNum=6"
    veraResponse = urllib.urlopen(url)
    states = json.load(veraResponse)['Device_Num_6']['states']

    for state in states:
        if state['variable'] == 'CurrentTemperature':
            return "Il fait "+state['value']+" degres"
< object

File: JARVIS-Engine/Plugins/Vera/lang/fr/modules/vera.py
# -*- coding: UTF-8 -*-
import urllib, json
from pymongo import MongoClient

class Vera:
    def __init__(self):
        self.configuration_jarvis = json.load(open('Configuration/jarvis.json'))
        mongo = MongoClient(self.configuration_jarvis['database']['server'], \
                            self.configuration_jarvis['database']['port'])
        self.configuration = mongo.jarvis.plugins.find_one({"name": "Vera"})

    def getTemperature(self, args):
        for device in self.configuration['configuration']['temperature']:
            url = "http://" + self.configuration['ip'] + ":" + self.configuration['port'] + \
                  "/data_request?id=status&output_format=json&DeviceNum=" + device['device_id']
            veraResponse = urllib.urlopen(url)
            states = json.load(veraResponse)['Device_Num_'+device['device_id']]['states']
        for state in states:
            for variable in device['variables']:
                if state['variable'] == variable['name']:
                    return json.dumps({"plugin": "vera","method": "getTemperature", \
                                       "body": u"Il fait "+state['value']+u" degrés"})